src/mango/src/mango_cursor.erl

```diff
@@ -1113,6 +1113,12 @@ extract_selector_hints_test_() ->
         ]
     }.
 
+expect_set_from_list_ordered(L) ->
+    case erlang:system_info(otp_release) of
+        N when N >= "28" -> lists:sort(L);
+        _ -> L
+    end.
+
```

Comment:
Instead of hard-coding OTP 28 checks in the test, the test should be improved to check that either the sets match or the the sorted lists of expected vs actual `unindexable_fields` match.

Reply:
I gave it a try here https://github.com/apache/couchdb/pull/5510. It's a general improvement regardless of OTP version that just make us use all new sets and sorts the results to they are deterministic, otherwise tests might fail randomly again in the future